enhancements/authentication/external-oidc-additional-identity-information-sources.md

```diff
@@ -0,0 +1,804 @@
+- **Identity Provider (IdP)** is an external piece of software responsible for creating access tokens used to authenticate requests against the Kubernetes API.
+- **Cluster User Identity** is an internal construct of Kubernetes for building the "identity" of a user that is often used for authorization and admission decisions.
+- **Cluster End User** is an individual using the cluster.
+
+#### Scenarios
+
+**Scenario**: Sourcing group information from an identity provider that was not provided in the JWT via the UserInfo endpoint.
+
+1. Cluster Administrator configures the `authentications.config.openshift.io/cluster` resource
+to tell OpenShift how to fetch the group information from the identity provider like so:
+```yaml
+apiVersion: config.openshift.io/v1
+kind: Authentication
+metadata:
+  name: cluster
+spec:
+  type: OIDC
+  oidcProviders:
+  - name: foo
+    claimMappings:
+      uid:
+        claim: email
+      username:
+        claim: username
+      groups:
+        claim: groups
+    externalClaims: # This is the proposed new field
+      clientAuth:
+        type: RequestProvidedToken # Use the token from the request as the access token to the UserInfo endpoint
+      claims:
+      - method: GET
+        url:
+          type: Static
+          static: "\"https://tenant.foo.io/userinfo\""
+        mappings:
+          - name: groups
+            expression: "has(response.groups) ? response.groups.join(',') : \"\""
+```
+2. End user authenticates with the IdP and gets an access token:
+
+```sh
+oc login --exec-plugin="oc-oidc" --client-secret=... --client-id=...
+```
+
+3. End users identify their group memberships:
+
+```sh
+oc apply -o yaml -f - <<EOF
+apiVersion: authentication.k8s.io/v1
+kind: SelfSubjectReview
+metadata:
+  name: test
+EOF
+```
+
+Output:
+```sh
+apiVersion: authentication.k8s.io/v1
+kind: SelfSubjectReview
+metadata:
+  creationTimestamp: "2025-12-09T17:03:01Z"
+status:
+  userInfo:
+    ...
+    groups:
+    - system:authenticated
+    - foo # fetched from UserInfo endpoint
+```
+
+**Scenario**: Sourcing group information from Entra ID that was not provided in the JWT via the Microsoft Graph API.
+
+1. Cluster Administrator configures the `authentications.config.openshift.io/cluster` resource
+to tell OpenShift how to fetch the group information from the identity provider like so:
+```yaml
+apiVersion: config.openshift.io/v1
+kind: Authentication
+metadata:
+  name: cluster
+spec:
+  type: OIDC
+  oidcProviders:
+  - name: foo
+    claimMappings:
+      uid:
+        claim: email
+      username:
+        claim: username
+      groups:
+        claim: groups
+    externalClaims: # This is the proposed new field
+      clientAuth:
+        type: ClientCredential # Use the token from the request as the access token to the UserInfo endpoint
+        clientCredential:
+          id: "kas"
+          secret: "kas-foo-client-secret"
```

Comment:
The comment is carried over from RequestProvidedToken -- secret is the name of a Secret resource containing the credential?